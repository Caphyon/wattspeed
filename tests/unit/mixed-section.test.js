import Vue from "vue";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import mixed_section from "../../src/components/sections/mixed-section.vue"; // name of your Vue component

global.EventBus = new Vue();
const localVue = createLocalVue();

const usingHttps = {
  mixedContent: {
    links: ['http://example.com', 'https://www.example.com'],
    srcs: [],
  },
  isHttps: true,
  data: {
    status: true,
    results: ['http://example.com'],
  },
  loading: false,
};

const notUsingHttps = {
  mixedContent: null,
  isHttps: false,
  loading: false,
};

const methods = {
  allMixed: jest.fn(),
};

function createWrapper(data) {
  return shallowMount(mixed_section, {
    localVue,
    methods,
    data() {
      return data;
    },
  });
}

describe("mixed-section.vue", () => {
  it("Testing mixed-section.vue component in case danger", () => {
    const wrapper = createWrapper(usingHttps);

    // controll if the component is instance
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.alert--danger').exists()).toBe(true);
    wrapper.destroy();
  });

  it("Testing mixed-section.vue component in case no mixed content", () => {
    usingHttps.data.status = false;
    usingHttps.data.results = [];
    const wrapper = createWrapper(usingHttps);

    // controll if the component is instance
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.alert--success').exists()).toBe(true);
    wrapper.destroy();
  });

  it("Testing mixed-section.vue component in case no https pages", () => {
    const wrapper = createWrapper(notUsingHttps);

    // controll if the component is instance
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.alert--warning').exists()).toBe(true);
    wrapper.destroy();
  });

  it('should return how many times the method is called inside the event bus', () => {
    const allMixedMock = jest.fn();
    const wrapper = createWrapper(usingHttps);
    wrapper.setMethods({ allMixed: allMixedMock });

    expect(wrapper.isVueInstance).toBeTruthy();
    global.EventBus.$emit('refreshData');

    // The mock function is called twice
    expect(allMixedMock.mock.calls.length).toBe(1);
    wrapper.destroy();
  });

});
