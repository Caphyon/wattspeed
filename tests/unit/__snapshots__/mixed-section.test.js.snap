// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixed-section.vue Testing mixed-section.vue component in case danger 1`] = `
<section-container-stub>
  <p
    class="alert--danger"
  >
    
        This site contains mixed content.
      
  </p>
</section-container-stub>
`;

exports[`mixed-section.vue Testing mixed-section.vue component in case no https pages 1`] = `
<section-container-stub>
  <p
    class="alert--warning"
  >
    
        Does not apply to non HTTPS pages.
    
  </p>
</section-container-stub>
`;

exports[`mixed-section.vue Testing mixed-section.vue component in case no mixed content 1`] = `
<section-container-stub>
  <p
    class="alert--success"
  >
    
        This page does not contain mixed content.
      
  </p>
</section-container-stub>
`;
